### PR TITLE
use strategy2 auth

### PR DESCRIPTION
**Pull request by steveburkett**
Move highlights to use asymmetric/symmetric auth for the user uuid cookie.

See also https://github.com/openstax/auth-rails/pull/1

https://app.zenhub.com/workspaces/openstax-unified-5b71aabe3815ff014b102258/issues/openstax/unified/763

### CODE REVIEW

File: config/secrets.yml.example [hunk 1]
```diff
@@ -2,19 +2,16 @@ development:
   secret_key_base: 517a4eb6769e6ce0399f4e345873ecd1d2d92b11e53b7ec0a281a6ceaa3762d627211d92d0de478198dad362417343993262c983c161004c47756ec1ff4511b8
   accounts:
     sso:
-      secret_key: <%= ENV["SSO_SHARED_SECRET"] ||
-                      '265127c36133669bedcf47f326e64e22623c1be35fffe04199f0d86bf45a3485' %>
-      secret_salt: <%= ENV["SSO_SHARED_SECRET_SALT"] || 'ox-shared-salt' %>
-      cookie_name: <%= ENV["SSO_COOKIE_NAME"] || 'ox' %>
```
Comment (jpslav) [inline config/secrets.yml.example:8]: pretty sure these are still needed to test strategy 1

File: config/secrets.yml.example [hunk 2]
```diff
@@ -31,6 +28,5 @@ production:
   admin_uuids: ssm(/external/highlights/admin_uuids)
   accounts:
     sso:
-      secret_key: ssm(sso_secret_key)
-      secret_salt: ssm(sso_secret_salt)
-      cookie_name: ssm(sso_cookie_name)
```
Comment (jpslav) [inline config/secrets.yml.example:36]: ditto above

**Review by jpslav — CHANGES_REQUESTED** [review]

File: Gemfile [hunk 3]
```diff
@@ -50,7 +50,8 @@ gem "openstax_swagger", github: 'openstax/swagger-rails', ref: '9bff4962b31e142d
 # Allow requests with a 'highlights' prefix to support use via CloudFront
 gem "openstax_path_prefixer", github: 'openstax/path_prefixer', ref: '0ed5cdba6be65dbf1d07fd7580e2311a2f42cdfd'
 
-gem "openstax_auth", github: 'openstax/auth-rails', ref: '1e4268f422bfa2562de3a8e57b6471cbb09f86ea'
+#gem "openstax_auth", github: 'openstax/auth-rails', ref: '1e4268f422bfa2562de3a8e57b6471cbb09f86ea'
```
Comment (jpslav) [inline Gemfile]: ready to remove?

**Review by jpslav — CHANGES_REQUESTED** [review]

File: Gemfile (hunk 3 above)
Comment (steveburkett) [reply Gemfile]: would like to make the CR fixes (for the auth gem part), test the full stack again, remove, then  retest the full stack again.  sound ok?

Comment (steveburkett) [reply Gemfile]: @jpslav i fixed this reference.  please +1 this if you think it's good.

Comment (jpslav) [reply Gemfile]: the commented out code is still here, that’s what I was referring to

**Review by jpslav — APPROVED** [review]

**Pull request merged**